### PR TITLE
Implemented yields

### PR DESCRIPTION
**Pull request by Vineet77**
I commented out tests not related to yields to allow for the yields test to be executed.

### CODE REVIEW

File: producer/src/contract/contract_test.go [hunk 1]
```diff
@@ -58,24 +75,60 @@ func TestMakeYield(t *testing.T) {
...
 
 /* Tests both serialization and deserialization */
 func TestYieldSerialization(t *testing.T) {
 	testPubKey := generatePubKey()
-	expected := MakeYield(testPubKey, 200000)
+	expected := MakeYield(&testPubKey, 200000)
 	serialized := expected.Serialize()
 	deserialized := DeserializeYield(serialized)
-	if !cmp.Equal(deserialized, expected) {
+	if !reflect.DeepEqual(deserialized, expected) {
```
Comment (kstola2) [inline producer/src/contract/contract_test.go]: Why was this changed? cmp.Equal is better for struct comparison than reflect

Comment (Vineet77) [reply producer/src/contract/contract_test.go]: For some reason reflect.DeepEqual works, and I also checked the values of the struct and they were equal, but cmp.Equal still returns the opposite.

Comment (Vineet77) [reply producer/src/contract/contract_test.go]: Also according to the document cmp.Equal is similar to reflect.DeepEqual.

Comment (kstola2) [reply producer/src/contract/contract_test.go]: If they are similar then they should both pass

Comment (Vineet77) [reply producer/src/contract/contract_test.go]: Yeah I think I made a mistake (probably missing a !) initially, because both ways work

**Comment by kstola2** [comment]
Looks good to me.
